app/models/project.rb

```diff
@@ -171,6 +171,10 @@ class Project < ApplicationRecord
   scope :with_member, ->(user = User.current) { where(id: user.memberships.select(:project_id)) }
   scope :without_member, ->(user = User.current) { where.not(id: user.memberships.select(:project_id)) }
 
+  scope :with_available_custom_fields, ->(custom_field_ids) do
```

Comment:
No unit test exists yet- would like to investigate using `Queries::Filters::Base#scope` to build the query entirely without touching the model.

At present, I'm not sure if that's very beneficial apart from reducing the footprint of this work.

Reply:
I think it is fine if we intend to use this scope outside of the queries, otherwise it would be better to try and implement it as part of the filter itself.